protostar/commands/test/test_environment_exceptions_test.py

```diff
@@ -102,6 +102,24 @@ def test_matching_by_partial_error_message():
     assert not RevertableException(error_message=["f", "b"]).match(ex)
 
 
+def test_get_metadata_by_type():
+    class UnusedMockMetadata(ExceptionMetadata):
+        @property
+        def name(self) -> str:
+            return "unused"
+
+        def format(self) -> str:
+            return "unused"
+
+    ex = SimpleReportedException("foo")
+
+    metadata = MockMetadata()
+    ex.metadata.append(metadata)
+
+    assert ex.get_metadata_by_type(MockMetadata) is metadata
+    assert ex.get_metadata_by_type(UnusedMockMetadata) is None
```

Comment:
What should happen, when somebody tries to add the same metadata twice?

Reply:
It's an undocumented programmer error. I will document that the same metadata type can only be added once.

Reply:
Done.